server/driver/driver.go

```diff
@@ -26,12 +26,15 @@ type Server interface {
 	// ListenAndServe listens on the TCP network address addr and then
 	// calls Serve with handler to handle requests on incoming connections.
 	// The addr argument will be a non-empty string specifying "host:port".
+	// If certFile or keyFile are non-empty, the Server should listen using
+	// TLS.
 	// The http.Handler will always be non-nil.
 	// Drivers must block until serving is done (or
 	// return an error if serving can't occur for some reason), serve
 	// requests to the given http.Handler, and be interruptable by Shutdown.
 	// Drivers should use the given address if they serve using TCP directly.
-	ListenAndServe(addr string, h http.Handler) error
+	ListenAndServe(addr, certFile, keyFile string, h http.Handler) error
```

Comment:
Why not a separate method to avoid breaking the API and to be more consistent with stdlib?

Reply:
Could do that; I considered it. I don't think it avoids breaking though; anyone who is implementing this interface will be broken whether we change an existing method or add a new one. The logic was largely the same so I went this way, but happy to change it if you prefer. LMK.

Reply:
(I haven't looked too closely at the rest of this PR, but...) If you add a different interface for `ListenAndServeTLS`, you can introduce this as a non-breaking change, and then have the `server` return an error for drivers that don't implement this optional interface.

Reply:
Also, separating the method saves two `""`s for user that just wants to use `ListenAndServe`.

Reply:
> If you add a different interface for ListenAndServeTLS, you can introduce this as a non-breaking change

True; I will give that a try. Although I still don't think it's likely that anyone is actually implementing this interface, it's good practice.

> separating the method saves two ""s for user that just wants to use ListenAndServe

The only user of the interface is the concrete type, so I don't think this is a big concern.

Reply:
Now using a separate / optional interface with a separate method. Thanks for the suggestion @zombiezen !

The PR is no longer marked as a breaking change.